### PR TITLE
fix: reduction of robot name fixes #4 and fixes #5

### LINKED ISSUE #4: Fix: Name of robot on Game page
Display another name of the robot on the game page. Maybe also the users can create their own name? Also fix positioning of the robot name, either below or above

### LINKED ISSUE #5: fix: Name of robot on Admin page
Name should here be same as Mac-address, but displayed either above or below

### PR DESCRIPTION
**Pull request by KajaKoren**
Just reduction of the Mac-address displayed to the user, having the whole MAC was quite messy. I assumed that 3 characters would be an ok amount to still distinguish between each of the robots.

### CODE REVIEW

File: src/components/Game/Robot.tsx [hunk 1]
```diff
@@ -165,7 +165,7 @@ export const Robot = ({
 				textAnchor="middle"
 				data-test-id="label"
 			>
-				{id}
+				{id.slice(0, 3)}
```
Comment (coderbyheart) [inline src/components/Game/Robot.tsx]: **suggestion**: turn this into a `shortId()` function, to ensure that the same transformation is applied everywhere and gets propagated properly when the requirements for having short ids changes.

The function then also is a central place to document why this is needed.

**Review by coderbyheart — CHANGES_REQUESTED** [review]

File: src/components/Game/Robot.tsx (hunk 1 above)
Comment (KajaKoren) [reply src/components/Game/Robot.tsx]: Not fully sure what you meant by the last sentence: "The function then also is a central place to document why this is needed". just add a description of why it is needed? 

I assume it would be appropriate to add a test for the function at least.

Comment (coderbyheart) [reply src/components/Game/Robot.tsx]: > Not fully sure what you meant by the last sentence: "The function then also is a central place to document why this is needed". just add a description of why it is needed?

Something like this:

```typescript

/**
 * This is needed, because ...
 */
const shortId = (id: string): string => id.slice(0,3)
```

> I assume it would be appropriate to add a test for the function at least.

Yes!

File: src/utils/shortId.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+/**This is needed, because we want to reduce the id/MAC-adress displayed for the
+ * robots to a given number that should be equal everywhere
+ */
```
Comment (coderbyheart) [inline src/utils/shortId.ts]:
```suggestion
/**
 * This is needed, because we want to reduce the id/MAC-adress displayed for the
 * robots to a given number that should be equal everywhere
 */
```

File: src/utils/shortId.spec.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { shortId } from 'utils/shortId.js'
+
+describe('shortId()', () => {
+	describe('it should reduce the string input to only the three first characters in the string ', () => {
+		it.each([
+			['Example', 'Exa'],
+			['Test', 'Tes'],
+			['robotId', 'rob'],
+			['Hello', 'Hel'],
```
Comment (coderbyheart) [inline src/utils/shortId.spec.ts]: Since it is going to be used to shorten ids, at least use one example Id here.

Also in your tests, think a bit about edge cases. What happens if the id is shorter than 3 characters? Right now all you tests are testing the same property of the function: shorting a string that is longer than 3 characters to a string that is exactly 3 characters.

**Review by coderbyheart — CHANGES_REQUESTED** [review]

File: src/components/Admin/RobotTeamAssigner.tsx [hunk 4]
```diff
@@ -27,7 +28,7 @@ export const RobotTeamAssigner = () => {
 						)
 						.map(([address, robot]) => (
 							<li key={address}>
-								{address}
+								{shortId(address) + ' '}
```
Comment (coderbyheart) [inline src/components/Admin/RobotTeamAssigner.tsx]:
```suggestion
								{shortId(address)}{' '}
```
This is the idiomatic react way of adding space.

File: src/components/Admin/RobotTeamAssigner.tsx [hunk 5]
```diff
@@ -59,7 +60,7 @@ const RobotTeam = ({
 
 	return (
 		<>
-			<label htmlFor={`robot-${address}-name`}>Team</label>
+			<label htmlFor={`robot-${address}-name`}>Team {':'}</label>
```
Comment (coderbyheart) [inline src/components/Admin/RobotTeamAssigner.tsx]:
```suggestion
			<label htmlFor={`robot-${address}-name`}>Team:</label>
```

Brackets not needed.

**Review by coderbyheart — APPROVED** [review]

**Comment by coderbyheart** [comment]
Looks good, some minor nitpicks.
